bluetooth: conn: Fix forced pairing request handling

raitraak-rrk opened the pull request:
BT_SECURITY_FORCE_PAIR option gets overridden when CONFIG_BT_SMP_SC_ONLY or CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY is defined. Cache the bit value before overrides.
Add explicit forced pair handling to existing security level check. Functionality did not change as this was done implicitly due to integer comparison for enums.
Add extra clarification to the method doc.
Signed-off-by: Rait Rääk <raitraak@gmail.com>

raitraak-rrk commented:
BT_SECURITY_FORCE_PAIR option within the bt_security_t feels like a hack that should not be there. Unfortunately, due to C limitations, to clean this up would be a breaking change.

jhedberg commented:
> BT_SECURITY_FORCE_PAIR option within the bt_security_t feels like a hack that should not be there. Unfortunately, due to C limitations, to clean this up would be a breaking change.

Yes. At the time the feature was added, backwards compatibility was precisely the reason for the chosen solution.

raitraak-rrk committed:
bluetooth: conn: Fix forced pairing request handling

BT_SECURITY_FORCE_PAIR option gets overridden when CONFIG_BT_SMP_SC_ONLY
or CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY is defined. Cache the bit value
before overrides.
Add explicit forced pair handling to existing security level check.
Functionality did not change as this was done implicitly due to integer
comparison for enums.
Add extra clarification to the method doc.
Signed-off-by: Rait Rääk <raitraak@gmail.com>
